deploy/docker/fs/opt/appsmith/entrypoint.sh

```diff
@@ -168,11 +168,32 @@ unset_unused_variables() {
   fi
 }
 
-check_mongodb_uri() {
-  echo "Checking APPSMITH_MONGODB_URI"
+configure_database_connection_url() {
+  echo "Configuring database connection URL"
+  isPostgresUrl=0
+  isMongoUrl=0
+  # Check if APPSMITH_DB_URL is not set
+  if [[ -z "${APPSMITH_DB_URL}" ]]; then
+    # If APPSMITH_DB_URL is not set, fall back to APPSMITH_MONGODB_URI
+    export APPSMITH_DB_URL="${APPSMITH_MONGODB_URI}"
+  fi
+
+  if [[ "${APPSMITH_DB_URL}" == *"postgresql"* ]]; then
+    isPostgresUrl=1
+    # Check if APPSMITH_DB_URL is a PostgreSQL URL and doesn't start with "jdbc:", prepend "jdbc:"
+    if [[ "${APPSMITH_DB_URL}" != "jdbc:"* ]]; then
+      export APPSMITH_DB_URL="jdbc:${APPSMITH_DB_URL}"
+    fi
+  elif [[ "${APPSMITH_DB_URL}" == *"mongodb"* ]]; then
+    isMongoUrl=1
+  fi
```

Comment:
Let's make this stricter and define a fallthrough `else` as well. Something like:

```suggestion
  if [[ "${APPSMITH_DB_URL}" == "jdbc:postgresql:"* ]]; then
    isPostgresUrl=1
  elif [[ "${APPSMITH_DB_URL}" == "postgresql:"* ]]; then
    isPostgresUrl=1
    export APPSMITH_DB_URL="jdbc:${APPSMITH_DB_URL}"
  elif [[ "${APPSMITH_DB_URL}" == "mongodb"* ]]; then
    isMongoUrl=1
  fi

```